src/preamble.js

```diff
@@ -1116,25 +1126,12 @@ try {
 var TOTAL_STACK = Module['TOTAL_STACK'] || {{{ TOTAL_STACK }}};
 var TOTAL_MEMORY = Module['TOTAL_MEMORY'] || {{{ TOTAL_MEMORY }}};
 
-var WASM_PAGE_SIZE = 64 * 1024;
-
-var totalMemory = WASM_PAGE_SIZE;
-while (totalMemory < TOTAL_MEMORY || totalMemory < 2*TOTAL_STACK) {
-  if (totalMemory < 16*1024*1024) {
-    totalMemory *= 2;
-  } else {
-    totalMemory += 16*1024*1024;
-  }
-}
-#if ALLOW_MEMORY_GROWTH
-totalMemory = Math.max(totalMemory, 16*1024*1024);
+if (TOTAL_MEMORY < 16*1024*1024) Module.printErr('TOTAL_MEMORY should be at least 16MB, was ' + TOTAL_MEMORY + '!');
```

Comment:
Actually, I'll remove these because those are things that asm.js/wasm compilation already checks, and no point just duplicating validation code here.